hazelcast/include/hazelcast/client/connection/AuthenticationFuture.h

```diff
@@ -30,37 +28,34 @@
 
 namespace hazelcast {
     namespace client {
-        class Address;
-
-        namespace spi {
-            class ClientContext;
+        namespace exception {
+            class IException;
         }
-
         namespace connection {
             class Connection;
 
-            class HAZELCAST_API OwnerConnectionFuture {
+            class HAZELCAST_API AuthenticationFuture {
             public:
-                OwnerConnectionFuture(spi::ClientContext&);
-
-                void markAsClosed();
+                AuthenticationFuture();
 
-                boost::shared_ptr<Connection> getOrWaitForCreation();
+                void onSuccess(const boost::shared_ptr<Connection> &connection);
 
-                boost::shared_ptr<Connection> createNew(const Address& address);
+                void onFailure(const boost::shared_ptr<exception::IException> &throwable);
 
-                void closeIfAddressMatches(const Address& address);
+                void onFailure(const exception::IException &throwable);
 
-                void close();
+                boost::shared_ptr<Connection> get();
             private:
-                boost::shared_ptr<Connection> ownerConnectionPtr;
-                spi::ClientContext& clientContext;
+                boost::shared_ptr<util::CountDownLatch> countDownLatch;
```

Comment:
Can you check if shared_ptr is necessary here ?

Reply:
yes, it is needed since it has an AtomicInt member which not copyable.